csrc/cpu/adam/cpu_adam.h

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0
+
+// DeepSpeed Team
+
+#pragma once
+
+#define NOMINMAX  // Windows idiosyncrasy
+                  // https://stackoverflow.com/questions/4913922/possible-problems-with-nominmax-on-visual-c
+
```

Comment:
This file has some duplication with `csrc/include/cpu_adam.h`. Do you plan to merge them?

Reply:
the CUDA version of cpu_adam.h provide Adam in the form of optimizer object.  We changed it to stateless function call to fit the scenario of FusedAdam.  Besides the CUDA version contains CUDA specific code, so we think its better to make a copy of cpu_adam.h to decouple.

Reply:
I understand. In the long run, it would be nice to remove duplicate code by having the CUDA cpu_adam inherit from the basic cpu_adam. But that will be a future PR.

Reply:
@delock, apologies for revisiting this issue. But I was reminded that we had made cpu_adam runnable in CPU only environment with PR #2507. Moreover, we created this [unit test](https://github.com/microsoft/DeepSpeed/blob/23a11a39510e2aefb48236e3d2672a7dcbfc42a3/tests/unit/ops/adam/test_cpu_adam.py#L90) for this scenario since `torch.optim.AdamW` can also work in cpu only environment. 

So I am unsure that the changes in `csrc/cpu` are actually needed. Did you try to use existing for to support CPU training? Thanks!

Reply:
Awesome! Thanks!